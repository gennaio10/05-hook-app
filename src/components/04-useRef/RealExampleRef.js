import React, { useState } from "react";
import "../02-useEffect/effects.css";
import { MultipleCustomHooks } from "../03-examples/MultipleCustomHooks";

export const RealExampleRef = () => {
  const [state, setState] = useState(true);

  return (
    <div>
      <h1>Real Example Ref</h1>
      {state && <MultipleCustomHooks />}
      <button className="btn btn-primary mt-5" onClick={()=>{setState(!state)}}>
          Show/Hide
      </button>
    </div>
  );
};
