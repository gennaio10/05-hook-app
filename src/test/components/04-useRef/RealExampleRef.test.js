import React from "react";
import { shallow } from "enzyme";
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";
import "@testing-library/jest-dom";

describe('Pruebas en <RealExampleRef />', () => {
    
    const wrapper = shallow(<RealExampleRef />);

    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
    })

    test('debe de no mostrar componente <MultipleCustomHooks />', () => {
        wrapper.find("button").simulate('click')
        expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);
    })
    
})
