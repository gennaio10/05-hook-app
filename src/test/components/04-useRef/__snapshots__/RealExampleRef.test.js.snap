// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RealExampleRef /> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <MultipleCustomHooks />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
